src/modules/string_utils.rs

```diff
@@ -162,6 +162,84 @@ impl StringUtils {
         })
     }
 
+    // $at (S n -- S')
+    #[cmd(name = "$at", stack)]
+    fn interpret_str_at(stack: &mut Stack) -> Result<()> {
+        let index = stack.pop_usize()?;
+        let string = stack.pop_string()?;
+
+        match string.chars().nth(index) {
+            Some(s) => stack.push(s.to_string()),
+            None => anyhow::bail!("index must be >= 0 and <= {}", usize::MAX),
+        }
+    }
+
+    // $mul (S n -- S*n)
+    #[cmd(name = "$mul", stack)]
+    fn interpret_str_mul(stack: &mut Stack) -> Result<()> {
+        let factor = stack.pop_usize()?;
+        let string = stack.pop_string()?;
+
+        stack.push(string.repeat(factor))
+    }
+
+    // $sybs (S -- t[S'0, S'1, S'2, ..., S'n])
+    #[cmd(name = "$sybs", stack)]
+    fn interpret_str_sybs(stack: &mut Stack) -> Result<()> {
+        let string = stack.pop_string()?;
+        let symbols = string
+            .chars()
+            .map(|c| Box::new(c.to_string()) as Box<dyn StackValue>)
+            .collect::<Vec<_>>();
+
+        stack.push(symbols)
+    }
+
+    // $sub (S x y -- S')
+    #[cmd(name = "$sub", stack)]
+    fn interpret_str_sub(stack: &mut Stack) -> Result<()> {
+        let y = stack.pop_usize()?;
+        let x = stack.pop_usize()?;
+        let string = stack.pop_string()?;
+
+        stack.push(string[x..y].to_string())
```

Comment:
Please add a bounds check here as it will panic by default

Reply:
FIXED